Demo/Demo/PayPalWebPayments/PayPalWebViewModel/PayPalWebViewModel.swift

```diff
@@ -73,22 +121,26 @@ class PayPalWebViewModel: ObservableObject {
                     let payPalRequest = PayPalWebCheckoutRequest(orderID: orderID, fundingSource: funding)
                     payPalWebCheckoutClient.start(request: payPalRequest) { result, error in
                         if let error {
-                            if error == PayPalError.checkoutCanceledError {
-                                print("Canceled")
-                                self.updateState(.idle)
-                            } else {
-                                self.updateState(.error(message: error.localizedDescription))
+                            DispatchQueue.main.async {
+                                if error == PayPalError.checkoutCanceledError {
+                                    print("Canceled")
+                                    self.state.createdOrderResponse = .idle
+                                } else {
+                                    self.state.createdOrderResponse = .error(message: error.localizedDescription)
```

Comment:
same here

Reply:
Since you created different responses for each stage or the order, you need to repurpose the result view or have different result views for each step.

Right now, you have a resultView that takes createdOrderResult

Reply:
Good catch . I will push a commit to refactor it.